Documentation Comments Style Guide - @throws

Parent issue: https://github.com/checkstyle/checkstyle/issues/19807

[Coverage table](https://checkstyle-diff-reports.s3.us-east-2.amazonaws.com/8c24aed_20260520164406/doc_comments_style.html#Coverage_table)

[Chapter link](https://www.oracle.com/technical-resources/articles/java/javadoc-tool.html#@exception)
## Content

### Image

<img width="1061" height="294" alt="Image" src="https://github.com/user-attachments/assets/56ee5359-6784-4130-8bf6-a36d82c20e0a" />

### Text
#### @throws (@exception was the original tag) (reference page)

A @throws tag should be included for any checked exceptions (declared in the throws clause), as illustrated below, and also for any unchecked exceptions that the caller might reasonably want to catch, with the exception of NullPointerException. Errors should not be documented as they are unpredictable. For more details, please see Documenting Exceptions with the @throws Tag.
```
/**
* @throws IOException  If an input or output 
*                      exception occurred
*/
public void f() throws IOException {
// body
}
```

See the Exceptions chapter of the Java Language Specification, Second Edition for more on exceptions. Also see order of multiple @throws tags.

## Guidelines
>A @throws tag should be included for any checked exceptions (declared in the throws clause)

This chapter is covered by [JavadocMethod](https://checkstyle.sourceforge.io/checks/javadoc/javadocmethod.html#JavadocMethod), using its property `validateThrows`. Proposal:

| Documentation Comments Rule | Checkstyle Checks Used | Sample files |
|---|---|---|
| `@throws` | ✅JavadocMethod | |

<img width="585" height="179" alt="Image" src="https://github.com/user-attachments/assets/a3b5af84-61cc-42ac-a0c5-5f50c4d47820" />

@gianmarcoschifone , 

> Errors should not be documented as they are unpredictable

we can not cover this for now, as we have no Check for this. Create new issue to think on this, Lets just mention this issue in coverage table and move forward. 

```
$ cat Test.java 
public interface Test {

        /**
         * Test method.
         *
         * @return some result
         * @throws Error sometime
         */
        String someMethod() throws OutOfMemoryError;
}

$ javadoc Test.java
Loading source file Test.java...
Constructing Javadoc information...
Building index for all the packages and classes...
Standard Doclet version 21.0.9+7-LTS-338
Building tree for all the packages and classes...
Generating ./Test.html...
Test.java:1: warning: no comment
public interface Test {
       ^
Generating ./package-summary.html...
Generating ./package-tree.html...
Generating ./overview-tree.html...
Building index for all classes...
Generating ./allclasses-index.html...
Generating ./allpackages-index.html...
Generating ./index-all.html...
Generating ./search.html...
Generating ./index.html...
Generating ./help-doc.html...
1 warning

```

<img width="851" height="717" alt="Image" src="https://github.com/user-attachments/assets/cdb3c3d5-a6e0-4be7-9327-1dfcedc2f542" />

--- a/config/jsoref-spellchecker/whitelist.words
+++ b/config/jsoref-spellchecker/whitelist.words
@@ -1390,6 +1390,7 @@ thomasjensen
 thr
 throwable
 throwscount
+throwstag
 Thymeleaf
 tidelift
 timeline

--- a/src/it/java/com/doccomments/checkstyle/test/writingdoccomments/tagconventions/throwstag/ThrowsTagTest.java
+++ b/src/it/java/com/doccomments/checkstyle/test/writingdoccomments/tagconventions/throwstag/ThrowsTagTest.java
@@ -1,0 +1,43 @@
+///////////////////////////////////////////////////////////////////////////////////////////////
+// checkstyle: Checks Java source code and other text files for adherence to a set of rules.
+// Copyright (C) 2001-2026 the original author or authors.
+//
+// This library is free software; you can redistribute it and/or
+// modify it under the terms of the GNU Lesser General Public
+// License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+//
+// This library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+// Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public
+// License along with this library; if not, write to the Free Software
+// Foundation, Inc., 59 Temple Place, Suite 330, Boston, MA  02111-1307  USA
+///////////////////////////////////////////////////////////////////////////////////////////////
+
+package com.doccomments.checkstyle.test.writingdoccomments.tagconventions.throwstag;
+
+import org.junit.jupiter.api.Test;
+
+import com.doccomments.checkstyle.test.base.AbstractDocCommentsModuleTestSupport;
+
+public class ThrowsTagTest extends AbstractDocCommentsModuleTestSupport {
+
+    @Override
+    public String getPackageLocation() {
+        return "com/doccomments/checkstyle/test/writingdoccomments/tagconventions/throwstag";
+    }
+
+    @Test
+    public void testCorrect() throws Exception {
+        verifyWithWholeConfig(getPath("InputCorrectThrowsTag.java"));
+    }
+
+    @Test
+    public void testIncorrect() throws Exception {
+        verifyWithWholeConfig(getPath("InputIncorrectThrowsTag.java"));
+    }
+
+}

--- a/src/it/resources/com/doccomments/checkstyle/test/writingdoccomments/tagconventions/requiredtags/InputRequiredTags.java
+++ b/src/it/resources/com/doccomments/checkstyle/test/writingdoccomments/tagconventions/requiredtags/InputRequiredTags.java
@@ -49,6 +49,7 @@ public class InputRequiredTags {
      *
      * @param value value to validate
      * @return validated value
+     * @throws IllegalArgumentException if value is negative
      */
     public int validate(int value) throws IllegalArgumentException {
         if (value < 0) {

--- a/src/it/resources/com/doccomments/checkstyle/test/writingdoccomments/tagconventions/throwstag/InputCorrectThrowsTag.java
+++ b/src/it/resources/com/doccomments/checkstyle/test/writingdoccomments/tagconventions/throwstag/InputCorrectThrowsTag.java
@@ -1,0 +1,51 @@
+package com.doccomments.checkstyle.test.writingdoccomments.tagconventions.throwstag;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+
+/**
+ * Input for correct {@code @throws} tag examples.
+ */
+public class InputCorrectThrowsTag {
+
+    /**
+     * Creates a sample instance.
+     */
+    public InputCorrectThrowsTag() {
+    }
+
+    /**
+     * Reads one byte from the source.
+     *
+     * @return byte value
+     * @throws IOException if reading fails
+     */
+    public int readByte() throws IOException {
+        return 1;
+    }
+
+    /**
+     * Opens the source file.
+     *
+     * @throws FileNotFoundException if the source file is missing
+     */
+    public void openSource() throws FileNotFoundException {
+    }
+
+    /**
+     * Closes the source file.
+     *
+     * @exception IOException if closing fails
+     */
+    public void closeSource() throws IOException {
+    }
+
+    /**
+     * Stores the supplied value.
+     *
+     * @param value value to store
+     * @throws IOException if storing fails
+     */
+    public void store(String value) throws IOException {
+    }
+}

--- a/src/it/resources/com/doccomments/checkstyle/test/writingdoccomments/tagconventions/throwstag/InputIncorrectThrowsTag.java
+++ b/src/it/resources/com/doccomments/checkstyle/test/writingdoccomments/tagconventions/throwstag/InputIncorrectThrowsTag.java
@@ -1,0 +1,33 @@
+package com.doccomments.checkstyle.test.writingdoccomments.tagconventions.throwstag;
+
+import java.io.FileNotFoundException;
+import java.io.IOException;
+
+/**
+ * Input for incorrect {@code @throws} tag examples.
+ */
+public class InputIncorrectThrowsTag {
+
+    /**
+     * Creates a sample instance.
+     */
+    public InputIncorrectThrowsTag() {
+    }
+
+    // violation 6 lines below 'Expected @throws tag for 'IOException'.'
+    /**
+     * Reads one byte without documenting the checked exception.
+     *
+     * @return byte value
+     */
+    public int missingThrowsTag() throws IOException {
+        return 1;
+    }
+
+    // violation 4 lines below 'Expected @throws tag for 'FileNotFoundException'.'
+    /**
+     * Opens a file without documenting the checked exception.
+     */
+    public void missingThrowsTagOnVoidMethod() throws FileNotFoundException {
+    }
+}

--- a/src/main/resources/doc_comments_checks.xml
+++ b/src/main/resources/doc_comments_checks.xml
@@ -39,10 +39,11 @@
     <!-- Tag conventions -->
     <module name="AtclauseOrder"/>
 
-    <!-- Required Tags --> <!-- @return -->
+    <!-- Required Tags --> <!-- @return --> <!-- @throws -->
     <module name="JavadocMethod">
       <property name="allowedAnnotations" value=""/>
       <property name="allowInlineReturn" value="true"/>
+      <property name="validateThrows" value="true"/>
     </module>
 
     <!-- Documenting default constructors -->

--- a/src/site/xdoc/doc_comments_style.xml
+++ b/src/site/xdoc/doc_comments_style.xml
@@ -734,8 +734,30 @@
                     @throws
                   </a>
                 </td>
-                <td>???</td>
-                <td/>
+                <td>
+                  <span class="wrapper inline">
+                    <img src="images/ok_blue.png" alt=""/>
+                  </span>
+                  <a href="checks/javadoc/javadocmethod.html#JavadocMethod">
+                    JavadocMethod</a>
+                  (<a href="https://github.com/search?q=path%3Asrc%2Fmain%2Fresources%20path%3A**%2Fdoc_comments_checks.xml+repo%3Acheckstyle%2Fcheckstyle+JavadocMethod">
+                  config</a>)
+                  <br/>
+                  <br/>
+                  <span class="wrapper inline">
+                    <img
+                        src="images/ban_red.png"
+                        alt="" />
+                  </span>
+                  Cannot cover the guideline that <code>Error</code> should not be documented,
+                    as Checkstyle does not have a Check for this yet. See
+                    <a href="https://github.com/checkstyle/checkstyle/issues/20792">#20792</a>.
+                </td>
+                <td>
+                  <a href="https://github.com/checkstyle/checkstyle/tree/master/src/it/resources/com/doccomments/checkstyle/test/writingdoccomments/tagconventions/throwstag">
+                    samples
+                  </a>
+                </td>
               </tr>
 
               <tr>

